### PR TITLE
Passing configuration options to JSON stream middleware

### PR DESCRIPTION
**Pull request by jpuri**
PR should be merged only after https://github.com/MetaMask/json-rpc-middleware-stream/pull/27 is deployed.

`json-rpc-middleware-stream` takes care to replay DAPP action when reconnection message is received. This PR passes the configuration parameter for the replay message in `json-rpc-middleware-stream`.

### CODE REVIEW

**Comment by mcmire** [comment]
It looks like the new version of `json-rpc-middleware-stream` drops support for Node 12, upgrading the `engines` field to `>=14.0.0`. We would need to do the same thing for this library to be able to use this version.

**Comment by jpuri** [comment]
PR is blocked by https://github.com/MetaMask/providers/pull/225

**Comment by jpuri** [comment]
The PR is ready for review @mcmire , @Gudahtt

File: src/StreamProvider.ts [hunk 1]
```diff
@@ -71,7 +71,9 @@ export abstract class AbstractStreamProvider extends BaseProvider {
     );
 
     // Set up RPC connection
-    this._jsonRpcConnection = createStreamMiddleware();
+    this._jsonRpcConnection = createStreamMiddleware({
+      retryOnMessage: 'METAMASK_EXTENSION_STREAM_CONNECT',
```
Comment (Gudahtt) [inline src/StreamProvider.ts:75]: Where is this message sent? Just trying to understand the whole solution.

Comment (jpuri) [reply src/StreamProvider.ts:75]: The message is sent from contentScript when extension stream is connected https://github.com/MetaMask/metamask-extension/blob/develop/app/scripts/contentscript.js#L395

**Review by Gudahtt — APPROVED** [review]
LGTM!

**Pull request merged**